kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/validation/JwsSignatureRecordValidationIT.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter.validation;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.InvalidRecordException;
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.kroxylicious.proxy.config.ConfigurationBuilder;
+import io.kroxylicious.proxy.config.NamedFilterDefinition;
+import io.kroxylicious.proxy.config.NamedFilterDefinitionBuilder;
+import io.kroxylicious.testing.kafka.api.KafkaCluster;
+import io.kroxylicious.testing.kafka.junit5ext.KafkaClusterExtension;
+import io.kroxylicious.testing.kafka.junit5ext.Topic;
+
+import static io.kroxylicious.test.jws.JwsTestUtils.ECDSA_VERIFY_JWKS;
+import static io.kroxylicious.test.jws.JwsTestUtils.JWS_HEADER_NAME;
+import static io.kroxylicious.test.jws.JwsTestUtils.RSA_AND_ECDSA_VERIFY_JWKS;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_ECDSA_JWK;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_ECDSA_JWK_AND_UNENCODED_CONTENT_DETACHED;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_ECDSA_JWK_AND_UNENCODED_CONTENT_DETACHED_PAYLOAD;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_ECDSA_JWK_PAYLOAD;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_MISSING_ECDSA_JWK;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_RSA_JWK;
+import static io.kroxylicious.test.jws.JwsTestUtils.VALID_JWS_USING_RSA_JWK_PAYLOAD;
+import static io.kroxylicious.test.tester.KroxyliciousConfigUtils.proxy;
+import static io.kroxylicious.test.tester.KroxyliciousTesters.kroxyliciousTester;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(KafkaClusterExtension.class)
+public class JwsSignatureRecordValidationIT extends RecordValidationBaseIT {
+    // Copied from JwsSignatureBytebufValidator
+    private static final String DEFAULT_ERROR_MESSAGE = "JWS Signature could not be successfully verified";
```

Comment:
If we really need it in the test just expose it, in package scope?, from the prod sources.